ovmi/lib/src/predicates/equal.rs

```diff
@@ -0,0 +1,54 @@
+use crate::executor::*;
+use crate::predicates::*;
+
+pub struct EqualPredicate<'a, Ext: ExternalCall> {
```

Comment:
This one looks like newtype patter, why didn't use [tuple struct](https://learning-rust.github.io/docs/b2.structs.html#Tuple-structs) here?

Reply:
Predicate may have arguments other than ext. For example, CompiledExecutable takes a constructor argument value. In other words, there is a bit chance that the code will change in the future.